pkg/machines/libvirt-dbus.js

```diff
@@ -800,11 +804,49 @@ const LIBVIRT_DBUS_PROVIDER = {
                             dispatch(updateVm(Object.assign({}, props, dumpxmlParams)));
                         else
                             dispatch(updateOrAddVm(Object.assign({}, props, dumpxmlParams)));
+
+                        dispatch(getVmSnapshots({ connectionName, domainPath: objPath }));
                     })
                     .catch(ex => console.warn("GET_VM action failed for path", objPath, ex.toString()));
         };
     },
 
+    GET_VM_SNAPSHOTS({ connectionName, domainPath }) {
+        return dispatch => call(connectionName, domainPath, 'org.libvirt.Domain', 'ListDomainSnapshots', [0], { timeout, type: 'u' })
+                .then((objPaths) => {
+                    const snaps = [];
+                    const promises = [];
+
+                    for (let i = 0; i < objPaths[0].length; i++) {
+                        const objPath = objPaths[0][i];
+                        promises.push(call(connectionName, objPath, 'org.libvirt.DomainSnapshot', 'GetXMLDesc', [0], { timeout, type: 'u' }));
+                    }
+
+                    // WA to avoid Promise.all() fail-fast behavior
+                    const toResultObject = (promise) => {
+                        return promise
+                                .then(result => ({ success: true, result }))
+                                .catch(error => ({ success: false, error }));
+                    };
+
+                    Promise.all(promises.map(toResultObject)).then(snapXmlList => {
+                        for (let i = 0; i < snapXmlList.length; i++) {
+                            if (snapXmlList[i].success) {
+                                const snapXml = snapXmlList[i].result[0];
+                                const dumpxmlParams = parseDomainSnapshotDumpxml(snapXml);
+                                snaps.push(dumpxmlParams);
+                            }
+                        }
+                        return dispatch(updateDomainSnapshots({
+                            connectionName,
+                            domainPath,
+                            snaps
+                        }));
+                    });
+                })
+                .fail(ex => console.warn("LIST_DOMAIN_SNAPSHOTS action failed for domain %s: %s", domainPath, JSON.stringify(ex)));
```

Comment:
Please don't mix the old jQuery promise (.done/.fail) and JS Promise (.then/.catch) syntax. Just use .catch().